skbio/sequence/_sequence.py

```diff
@@ -538,9 +586,13 @@ def _assert_can_cast_to(cls, target):
     def _positional_metadata_axis_len_(self):
         return len(self)
 
+    @overrides(IntervalMetadataMixin)
+    def _interval_metadata_axis_len_(self):
+        return len(self)
+
     @stable(as_of="0.4.0")
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 lowercase=False):
+                 interval_metadata=None, lowercase=False):
```

Comment:
@ebolyen and I chatted some more and are on the fence with this one. We are okay with leaving the order as-is (`sequence, metadata=None, positional_metadata=None, interval_metadata=None, lowercase=False`) because it is unlikely in practice to break user code. A user's code would break only if they had supplied `metadata`, `postional_metadata`, *and* `lowercase` parameters positionally. In the unlikely event that this happens, users will get an error telling them a `bool` isn't a valid `IntervalMetadata` type, so it won't silently produce buggy behavior.

Can you add a note under the `Backward-incompatible changes [stable]` heading for 0.5.0-dev changelog? Please provide a detailed explanation of this change, including the previous constructor signature and the new one, why the change was made, and why we are okay with breaking stable API (just transcribe what I wrote above if you want). Thanks!